kythe/go/serving/xrefs/xrefs.go

```diff
@@ -93,22 +93,67 @@ type SplitTable struct {
 
 	// Documentation is a table of srvpb.Documents keyed by their node ticket.
 	Documentation table.Proto
+
+	// RewriteEdgeLabel is an optional callback to rewrite edge labels.
+	// It will be called once per request; the function it returns will then be
+	// called once per edge.
+	RewriteEdgeLabel func(context.Context) func(*string)
```

Comment:
Why is the callback func(*string) instead of func(string) string?

Reply:
The rewrite* functions all rewrite messages and fields in place.

Reply:
Where are the existing rewrite functions?

Reply:
They're added in this PR. If something needs to be changed please let me know.

Reply:
Sorry, your answer to my question made it sound like you didn't have control over the rewrite functions or were following existing behavior.

Why are you rewriting in place instead of assigning a new value to the field? It seems easier for clients to write a function that goes from string to string than change the contents of an existing string.

Reply:
No worries. I've changed the type in the most recent push.